components/popover_list_members/index.ts

```diff
@@ -2,27 +2,50 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
 
 import {createSelector} from 'reselect';
 
 import {getAllChannelStats} from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
 import {getCurrentUserId, getUserStatuses, makeGetProfilesInChannel} from 'mattermost-redux/selectors/entities/users';
 import {getTeammateNameDisplaySetting, getAddMembersToChannel} from 'mattermost-redux/selectors/entities/preferences';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Channel} from 'mattermost-redux/types/channels';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+
+import {GlobalState} from 'types/store';
+import {ModalData} from 'types/actions';
 
 import {openDirectChannelToUserId} from 'actions/channel_actions.jsx';
 import {loadProfilesAndStatusesInChannel} from 'actions/user_actions.jsx';
 import {openModal} from 'actions/views/modals';
+
 import {canManageMembers} from 'utils/channel_utils';
 import {sortUsersByStatusAndDisplayName} from 'utils/utils.jsx';
 
-import PopoverListMembers from './popover_list_members.jsx';
+import PopoverListMembers, {Props} from './popover_list_members';
+
+type Filters = {
+    role?: string;
+    inactive?: boolean;
+    active?: boolean;
+    roles?: string[];
+    exclude_roles?: string[];
+    channel_roles?: string[];
+    team_roles?: string[];
+};
+
+type Actions = {
+    openModal: <P>(modalData: ModalData<P>) => void;
+    loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage: number | undefined, sort: string, options: {active: boolean}) => void;
```

Comment:
out of scope for this PR, but I prefer to have optional parameters (in this case `perPage`) coming after the mandatory ones.

```suggestion
    loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage?: number, sort: string, options: {active: boolean}) => void;
```

Reply:
I reverted this because when I make `perPage` optional as you said I need to put after required ones due to TypeScript rule. When I do this, I mix the order in [UserActions](https://github.com/mattermost/mattermost-webapp/blob/1ab8a75409c4ea71844fa92a8f9fdf2f748f52bd/actions/user_actions.jsx#L39). Any idea how to address this without changing  `UserActions`?

Reply:
in that case we could just leave it as is, as the person writing the `UserAction` was not paying attention. I will create a follow-up ticket for this to re-visit this in the future.

Reply:
well... i should have looked at the function definition before writing that comment ... it is written as it should be: all of the optional params are at the end.

just type it as follows:

```
loadProfilesAndStatusesInChannel: (channelId: string, page?: number, perPage?: number, sort?: string, options?: {active: boolean}) => void;
```

Reply:
Ahh I see your point. As all params except `channelId` have default value, we can mark them optional with `?`.  Done